packages/next/src/next-devtools/dev-overlay/components/errors/error-overlay-toolbar/nodejs-inspector-button.tsx

```diff
@@ -248,41 +233,85 @@ function NodeJsDisabledIcon(props: any) {
   )
 }
 
-const label =
-  'Learn more about enabling Node.js inspector for server code with Chrome DevTools'
-
 export function NodejsInspectorButton({
-  devtoolsFrontendUrl,
+  defaultDevtoolsFrontendUrl,
 }: {
-  devtoolsFrontendUrl: string | undefined
+  defaultDevtoolsFrontendUrl: string | undefined
 }) {
-  const content = devtoolsFrontendUrl || ''
-  const disabled = !content || !isChromeBrowser
-  if (disabled) {
+  const [devtoolsFrontendUrlState, attachDebuggerAction, isAttachingDebugger] =
+    useActionState<
+      | { status: 'fulfilled'; value: string | undefined }
+      | { status: 'rejected'; reason: unknown }
+    >(
+      async () => {
+        try {
+          const response = await fetch('/__nextjs_attach-nodejs-inspector', {
+            method: 'POST',
+          })
+          if (!response.ok) {
+            throw new Error(
+              `${response.status} ${response.statusText}: ${await response.text()}`
+            )
+          }
+          const devtoolsFrontendUrl = await response.json()
+          return {
+            status: 'fulfilled',
+            value: devtoolsFrontendUrl,
+          }
+        } catch (cause) {
+          return {
+            status: 'rejected',
+            reason: new Error(
+              'Failed to attach Node.js inspector: ' +
+                // TODO: Use `cause` property once Redbox supports displaying `cause`
+                String(cause)
+            ),
+          }
+        }
+      },
+      { status: 'fulfilled', value: defaultDevtoolsFrontendUrl }
+    )
+
+  const devtoolsFrontendUrl =
+    devtoolsFrontendUrlState.status === 'fulfilled'
+      ? devtoolsFrontendUrlState.value
+      : undefined
+
+  useEffect(() => {
+    if (devtoolsFrontendUrlState.status === 'rejected') {
+      console.error(devtoolsFrontendUrlState.reason)
+    }
+  }, [devtoolsFrontendUrlState])
+
+  const attachDebugger = startTransition.bind(null, attachDebuggerAction)
```

Comment:
Just out of curiosity: Did you decide to use this pattern to avoid rendering a `<form>` element around the button?